parsing.go

```diff
@@ -18,7 +19,15 @@ const maxRequestSize = 16 * 1024 * 1024 // 16 MB
 // passed to the method in the URL. Use this helper function when
 // writing handler functions.
 func GetVars(r *http.Request) map[string]string {
-	return mux.Vars(r)
+	vars := mux.Vars(r)
+	for k, v := range vars {
+		unescapedVar, err := url.PathUnescape(v)
+		if err != nil {
+			return nil
```

Comment:
Is the correct move here to return a nil map? It seems like this could easily cause a panic since code typically expects `GetVars` to return a non-nil map, even if it's empty.